### PR TITLE
Add certified moderator badge

### PR DESCRIPTION
**Pull request by webtax-gh**
This pull requests adds the new certified moderator user flag.

PR in discord-api-docs: discord/discord-api-docs#2946

### CODE REVIEW

File: include/dpp/user.h [hunk 1]
```diff
@@ -65,7 +65,9 @@ enum user_flags {
 	/// User has the Early Verified Bot Developer badge
 	u_verified_bot_dev =	0b00000100000000000000000,
 	/// User's icon is animated
-	u_animated_icon =	0b00001000000000000000000
+	u_animated_icon =	0b00001000000000000000000,
+	/// User is a certified moderator
+	u_certified_moderator =	0b00010000000000000000000
```
Comment (Lulalaby) [inline include/dpp/user.h:70]: Change to `u_discord_certified_moderator`

File: include/dpp/user.h [hunk 2]
```diff
@@ -217,6 +219,12 @@ class user : public managed {
 	 * @return true if verified bot developer
 	 */
 	bool is_verified_bot_dev() const;
+	/**
+	 * @brief Return true if user is a certified moderator
+	 * 
+	 * @return true if certified moderator
+	 */
+	bool is_certified_moderator() const;
```
Comment (Lulalaby) [inline include/dpp/user.h:227]: Change to `is_discord_certified_moderator()`

File: src/dpp/user.cpp [hunk 3]
```diff
@@ -38,7 +38,8 @@ std::map<uint32_t, dpp::user_flags> usermap = {
 	{ 1 << 10,      dpp::u_team_user },
 	{ 1 << 14,      dpp::u_bughunter_2 },
 	{ 1 << 16,      dpp::u_verified_bot },
-	{ 1 << 17,      dpp::u_verified_bot_dev }
+	{ 1 << 17,      dpp::u_verified_bot_dev },
+	{ 1 << 18,	dpp::u_certified_moderator }
```
Comment (Lulalaby) [inline src/dpp/user.cpp:42]: Change to `dpp::u_discord_certified_moderator`

File: src/dpp/user.cpp [hunk 4]
```diff
@@ -139,6 +140,10 @@ bool user::is_verified_bot_dev() const {
 	 return this->flags & u_verified_bot_dev; 
 }
 
+bool user::is_certified_moderator() const {
```
Comment (Lulalaby) [inline src/dpp/user.cpp:143]: Change to `user::is_discord_certified_moderator()`

File: src/dpp/user.cpp [hunk 5]
```diff
@@ -139,6 +140,10 @@ bool user::is_verified_bot_dev() const {
...
+	 return this->flags & u_certified_moderator;
```
Comment (Lulalaby) [inline src/dpp/user.cpp:144]: Change to `u_discord_certified_moderator`

File: include/dpp/user.h [hunk 6]
```diff
@@ -217,6 +219,12 @@ class user : public managed {
 	 * @return true if verified bot developer
 	 */
 	bool is_verified_bot_dev() const;
+	/**
+	 * @brief Return true if user is a certified moderator
```
Comment (Lulalaby) [inline include/dpp/user.h:223]: `is a discord certified moderator`

File: include/dpp/user.h [hunk 7]
```diff
@@ -217,6 +219,12 @@ class user : public managed {
...
+	 * 
+	 * @return true if certified moderator
```
Comment (Lulalaby) [inline include/dpp/user.h:225]: `discord certified moderator`

**Review by Lulalaby — COMMENTED** [review]
Fix things do to the name change to ``Discord Certified Moderator``

**Comment by braindigitalis** [comment]
isn't it obvious where the user is a certified moderator? it's discord after all, isn't this just more typing for typings sake?

**Comment by Lulalaby** [comment]
> 
> 
> isn't it obvious where the user is a certified moderator? it's discord after all, isn't this just more typing for typings sake?

No it is related to the DMD (Discord Moderator Discord).
And the Discord Staff assigns it there.
It's like the Discord Staff Badge
It indicates that they're official approved by Discord.
I think it's important to name it right.

**Comment by webtax-gh** [comment]
The client internally references it as `certified_moderator` so i don't see your point.

**Comment by Lulalaby** [comment]
image
You should stick to the docs.

**Comment by webtax-gh** [comment]
I agree with Brain when he said it's just typing for typing's sake. We dont need our thing to 100% match the docs, and keep in mind the docs might change. Imo shorter is better. I'm locking this conversion to contributors now.

**Pull request merged**